### PR TITLE
cherry-pick: pool operation fixes from develop

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
CP fixes to release/2.0

### CODE REVIEW

**Comment by Abhinandan-Purkait** [comment]
bors try

File: control-plane/agents/src/bin/core/controller/wrapper.rs [hunk 1]
```diff
@@ -972,13 +975,29 @@ impl ClientOps for Arc<tokio::sync::RwLock<NodeWrapper>> {
...
+    async fn import_pool(&self, request: &ImportPool) -> Result<PoolState, SvcError> {
+        let dataplane = self.grpc_client_locked(request.id()).await?;
+        let import_response = dataplane.import_pool(request).await;
+        let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
+        self.update_pool_states(ctx.deref_mut()).await?;
+        self.update_replica_states(ctx.deref_mut()).await?;
+        let pool = self.read().await.pool(&request.id);
+        match (pool, import_response) {
+            (_, Ok(pool)) => Ok(pool),
+            (_, Err(error)) => Err(error),
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/wrapper.rs]: seems we're never using pool that we got from self?

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/agents/src/bin/core/controller/wrapper.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/controller/wrapper.rs]: Will add the AlreadyExists arm to consume it.

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors cancel

File: control-plane/agents/src/bin/core/controller/wrapper.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/controller/wrapper.rs]: Done.

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors cancel

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors ping

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors try

**Comment by avishnu** [comment]
bors try

**Review by avishnu — APPROVED** [review]

**Comment by avishnu** [comment]
bors try

**Comment by avishnu** [comment]
bors try-

**Comment by avishnu** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**